docs/source/pruning.md

```diff
@@ -386,6 +386,37 @@ Please refer to [pruning examples](../../examples/README.md#Pruning-1) for more
 
 Particular hardware/software like [Intel Extension for Transformer](https://github.com/intel/intel-extension-for-transformers) are required to obtain inference speed and footprints' optimization for most sparse models. However, using [model slim](#click) for some special structures can obtain significant inference speed improvements and footprint reduction without the post-pruning deployment. In other words, you can achieve model acceleration directly under your training framework (PyTorch, etc.)
 
+## Pruning with Hyperparameter Optimization
+### Reason for HPO
+1. It reduces the human effort required, since many ML developers spend considerable time tuning the
+hyper-parameters, especially for large datasets or complex ML algorithms with a large number of
+hyperparameters.
+2. It improves the performance of ML models. Many ML hyper-parameters have different optimums to
+achieve best performance in different datasets or problems.
+3. It makes the models and research more reproducible. Only when the same level of hyper-parameter
+tuning process is implemented can different ML algorithms be compared fairly; hence, using a same HPO
+method on different ML algorithms also helps to determine the most suitable ML model for a specific
+problem
+
+### Get Started with HPO API
```

Comment:
move this readme to the HPO folder and add a concise summary in pruning.md. there are two pruning.md, the other one is in the pruner folder, please help sync them

Reply:
done